core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -303,6 +287,325 @@ public static Properties filterAndRemovePrefix(Map<String, String> properties, S
     return result;
   }
 
+  static String getTableOrViewSql(boolean table) {
```

Comment:
I think we can split this one into `getTableSql()` and `getViewSql()` like other package private methods we have now.

Reply:
The reason of package visibility is the test. That said, I can use `getTableSql()` and `getViewSql()` in the tests. I will do that.

Reply:
+1 on having package-private `getTableSql()` / `getViewSql()`. Also I would rename `boolean table` to `boolean isTable`everywhere